diff --git a/lib/pc.js b/lib/pc.js
index 332f542b78b90a6dcb4eb6569169457ec0826592..1e457bd67e838b5e393d479bcda447dcd8980125 100644
--- a/lib/pc.js
+++ b/lib/pc.js
@@ -95,7 +95,7 @@ var _sfc_main = /* @__PURE__ */ defineComponent({
     });
   },
   render: function render() {
-    var _state$formInstance, _state$formInstance2, _state$formInstance3, _state$formInstance4, _this = this, _class, _class2, _class3, _class4, _class5;
+    var _state$formInstance, _state$formInstance2, _state$formInstance3, _state$formInstance4, _state$formInstance5, _this = this, _class, _class2, _class3, _class4, _class5;
     var state = this.state, required = this.required, slots = this.slots, label = this.label, scopedSlots = this.scopedSlots, showMessage = this.showMessage, inlineMessage = this.inlineMessage, ellipsis = this.ellipsis, vertical = this.vertical, handleLabelMouseenter = this.handleLabelMouseenter, handleMouseleave = this.handleMouseleave;
     var validateIcon = state.validateIcon, isErrorInline = state.isErrorInline, isErrorBlock = state.isErrorBlock, tooltipType = state.tooltipType;
     var isMobile = state.mode === "mobile";
@@ -111,7 +111,7 @@ var _sfc_main = /* @__PURE__ */ defineComponent({
     var formAppendToBody = (_state$formInstance3 = state.formInstance) == null ? void 0 : _state$formInstance3.appendToBody;
     var appendToBody = typeof this.appendToBody === "boolean" ? this.appendToBody : typeof formAppendToBody === "boolean" ? formAppendToBody : true;
     var validatePosition = this.validatePosition || ((_state$formInstance4 = state.formInstance) == null ? void 0 : _state$formInstance4.validatePosition) || "top-end";
-    var popperOptions = _extends({}, state.formInstance.popperOptions, this.popperOptions, {
+    var popperOptions = _extends({}, (_state$formInstance5 = state.formInstance) == null ? void 0 : _state$formInstance5.popperOptions, this.popperOptions, {
       forceAbsolute: !appendToBody,
       onUpdate: function onUpdate(options) {
         var popper = options.instance._popper;
